lms/views/dashboard/api/assignment.py

```diff
@@ -142,16 +142,11 @@ def course_assignments_metrics(self) -> APIAssignments:
         course = self.dashboard_service.get_request_course(
             self.request, self.request.matchdict["course_id"]
         )
-        course_students = self.request.db.scalars(
-            self.user_service.get_users(
-                course_ids=[course.id],
-                role_scope=RoleScope.COURSE,
-                role_type=RoleType.LEARNER,
-                instructor_h_userid=current_h_userid,
-                admin_organization_ids=[org.id for org in admin_organizations],
-                h_userids=filter_by_h_userids,
-            )
-        ).all()
+        _, course_students_query = self.dashboard_service.get_course_roster(
```

Comment:
Instead of making a generic get_users query, use the dashboard service method that returns roster data if available.